### PR TITLE
DLP: Added sample for de-identify using time extract

### PR DESCRIPTION
**Pull request by dinesh-crest**
DLP: Added sample for de-identify using time extract
Added unit test cases for the same

## Description

Reference:- https://cloud.google.com/dlp/docs/transformations-reference#time-extract

Note: Before submitting a pull request, please open an issue for discussion if you are not associated with Google.

## Checklist
- [x] I have followed guidelines from [CONTRIBUTING.MD](https://github.com/GoogleCloudPlatform/nodejs-docs-samples/blob/main/CONTRIBUTING.md) and [Samples Style Guide](https://googlecloudplatform.github.io/samples-style-guide/)
- [x] **Tests** pass:   `npm test` (see [Testing](https://github.com/GoogleCloudPlatform/nodejs-docs-samples/blob/main/CONTRIBUTING.md#run-the-tests-for-a-single-sample))
- [x] **Lint** pass:   `npm run lint` (see [Style](https://github.com/GoogleCloudPlatform/nodejs-docs-samples/blob/main/CONTRIBUTING.md#style))
- [ ] These samples need a new **API enabled** in testing projects to pass (let us know which ones)
- [ ] These samples need a new/updated **env vars** in testing projects set to pass (let us know which ones)
- [ ] This sample adds a new sample directory, and I updated the [CODEOWNERS file](https://github.com/GoogleCloudPlatform/nodejs-docs-samples/blob/main/CODEOWNERS) with the codeowners for this sample
- [ ] This sample adds a new sample directory, and I created [GitHub Actions workflow](https://github.com/GoogleCloudPlatform/nodejs-docs-samples/blob/main/CONTRIBUTING.md#adding-new-samples) for this sample
- [ ] This sample adds a new **Product API**, and I updated the [Blunderbuss issue/PR auto-assigner](https://github.com/GoogleCloudPlatform/nodejs-docs-samples/blob/main/.github/blunderbuss.yml) with the codeowners for this sample
- [x] Please **merge** this PR for me once it is approved

### CODE REVIEW

**Review by kweinmeister — APPROVED** [review]

**Comment by pattishin** [comment]
Hi @dinesh-crest! I was diving into the failed workflow for your PRs, due to the way we've set the authentication up - you may need to directly clone this repo and create a PR off of it rather than your forked version.

**Comment by dinesh-crest** [comment]
> Hi @dinesh-crest! I was diving into the failed workflow for your PRs, due to the way we've set the authentication up - you may need to directly clone this repo and create a PR off of it rather than your forked version to avoid the current failure.

hi @pattishin This was discussed earlier with @annadata and we decided to go with this approach as I don't have enough permissions like creating a branch using the direct clone approach. Maybe @annadata can throw some light on this

File: dlp/deidentifyWithTimeExtraction.js [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+  // Instantiates a client
+  const dlp = new DLP.DlpServiceClient();
+
+  // The project ID to run the API call under
+  // const projectId = 'my-project';
+
+  // The string to de-identify
+  // const string = 'My BirthDay is on 9/21/1976';
+
+  async function deidentifyWithTimeExtraction() {
```
Comment (annadata) [inline dlp/deidentifyWithTimeExtraction.js:35]: Please align the example with https://github.com/GoogleCloudPlatform/java-docs-samples/blob/main/dlp/snippets/src/main/java/dlp/snippets/DeIdentifyWithTimeExtraction.java
It uses a table.

**Review by annadata — COMMENTED** [review]
LGTM otherwise.

**Review by annadata — APPROVED** [review]

**Pull request merged**